### PR TITLE
Addressing UNIX problems with Queue.qsize()

### PR DESCRIPTION
**Pull request by vigji**

### CODE REVIEW

File: arrayqueues/portable_queue.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        with self.count.get_lock():
+            self.count.value += n
+
+    @property
+    def value(self):
+        """ Return the value of the counter """
+        return self.count.value
+
+
+class Queue(multiprocessing.queues.Queue):
```
Comment (vilim) [inline arrayqueues/portable_queue.py]: I think it is just under multiprocessing

**Review by vilim — CHANGES_REQUESTED** [review]
As soon as Travis is happy, we are ready to merge. It would be good to add the automated release infrastructure to this repo as well

File: arrayqueues/portable_queue.py (hunk 1 above)
Comment (vigji) [reply arrayqueues/portable_queue.py]: it looks like the importing ways from multiprocessing can do funny things to the multiprocessing context, that's why I kept it that way

**Comment by vigji** [comment]
I thought it was as easy as this, but now I struggle with the following problem: if I set up an `ArrayQueue()` as process attribute, after I `start()` the process the `queue` object attribute of the `arrayqueue` does not have `.size` anymore (after having been copied in the separate process maybe?)

**Comment by vigji** [comment]
eg from sashimi:

```
class VolumeDispatcher(LoggingProcess):
    def __init__(
        self,
        stop_event: LoggedEvent,
        saving_signal: LoggedEvent,
        wait_signal: LoggedEvent,
        noise_subtraction_on: Event,
        camera_queue: ArrayQueue,
        saver_queue: ArrayQueue,
        max_queue_size=1200,
    ):
        super().__init__(name="dispatcher")
        self.stop_event = stop_event
        self.saving_signal = saving_signal.new_reference(self.logger)
        self.wait_signal = wait_signal.new_reference(self.logger)
        self.noise_subtraction_active = noise_subtraction_on.new_reference(self.logger)

        self.camera_queue = camera_queue
        self.saver_queue = saver_queue
        self.n_planes_queue = Queue()
        self.viewer_queue = ArrayQueue(max_mbytes=max_queue_size)
        self.calibration_ref_queue = ArrayQueue()
        print(self.calibration_ref_queue.queue.size)

        self.volume_buffer = None
        self.current_frame = None
        self.calibration_ref = None

        self.n_planes = 1
        self.i_plane = 0
        self.first_volume = True

    def run(self):
        print(self.calibration_ref_queue.queue.size)
```

The second print fails

**Comment by vigji** [comment]
for reference: https://stackoverflow.com/questions/39496554/cannot-subclass-multiprocessing-queue-in-python-3-5

**Comment by vigji** [comment]
The last commit should fix it, but it still mysteriously fails even locally pytests with the following error:

`ERROR arrayqueues/tests/test_shared_arrays.py - AttributeError: module 'multiprocessing' has no attribute 'queues'`

Which does not happen in normal execution...

**Comment by vilim** [comment]
Python version?

**Comment by vigji** [comment]
> Python version?

Python 3.8.5

**Comment by vilim** [comment]
What I am saying is that we can put travis to 3.7 and 3.8, no need to support 3.6 anymore

**Comment by vigji** [comment]
I don't get your point, `pytest` is failing locally on my computer with python 3.8 b/c of that error

**Comment by vilim** [comment]
ah sorry, misread.

**Comment by vigji** [comment]
Seems to be working now

**Pull request merged**